### PR TITLE
redirect inflation

### PR DESCRIPTION
**Pull request by deckb**
## Change Description

Create a contract to distribute inflation coming from the eosio contract action `setinflation`. The idea is to enable multiple accounts to receive inflation including `eosio.rex` via a new action `donatetorex`.

## Deployment Changes
- [X ] Deployment Changes

Creation of `eosio.dist` account is required if the contract is built with `USE_INFLATION_DISTRIBUTE`

## API Changes
- [ ] API Changes

## Documentation Additions
- [ X] Documentation Additions

Add additional contract `eosio.distribute` to the docs.

### CODE REVIEW

File: contracts/eosio.system/include/eosio.system/eosio.system.hpp [hunk 1]
```diff
@@ -687,14 +687,19 @@ namespace eosiosystem {
...
-         static constexpr eosio::name saving_account{"eosio.saving"_n};
          static constexpr eosio::name rex_account{"eosio.rex"_n};
          static constexpr eosio::name reserv_account{"eosio.reserv"_n};
          static constexpr eosio::name null_account{"eosio.null"_n};
          static constexpr symbol ramcore_symbol = symbol(symbol_code("RAMCORE"), 4);
          static constexpr symbol ram_symbol     = symbol(symbol_code("RAM"), 0);
          static constexpr symbol rex_symbol     = symbol(symbol_code("REX"), 4);
 
+         #ifdef USE_INFLATION_DISTRIB
+            const inline static name inflation_account = "eosio.ditrib"_n;
```
Comment (DenisCarriere) [inline contracts/eosio.system/include/eosio.system/eosio.system.hpp]: Perhaps using `eosio.dist` as the abbreviated EOSIO account name

File: contracts/eosio.system/include/eosio.system/eosio.system.hpp [hunk 2]
```diff
@@ -687,14 +687,19 @@ namespace eosiosystem {
...
          static constexpr eosio::name names_account{"eosio.names"_n};
-         static constexpr eosio::name saving_account{"eosio.saving"_n};
          static constexpr eosio::name rex_account{"eosio.rex"_n};
          static constexpr eosio::name reserv_account{"eosio.reserv"_n};
          static constexpr eosio::name null_account{"eosio.null"_n};
          static constexpr symbol ramcore_symbol = symbol(symbol_code("RAMCORE"), 4);
          static constexpr symbol ram_symbol     = symbol(symbol_code("RAM"), 0);
          static constexpr symbol rex_symbol     = symbol(symbol_code("REX"), 4);
 
+         #ifdef USE_INFLATION_DISTRIB
```
Comment (DenisCarriere) [inline contracts/eosio.system/include/eosio.system/eosio.system.hpp]: flag could be defined in full

`USE_INFLATION_DISTRIBUTE`

File: contracts/eosio.system/include/eosio.system/eosio.system.hpp [hunk 3]
```diff
@@ -1287,7 +1298,7 @@ namespace eosiosystem {
           *     (eg. For 5% Annual inflation => annual_rate=500
           *          For 1.5% Annual inflation => annual_rate=150
           * @param inflation_pay_factor - Inverse of the fraction of the inflation used to reward block producers.
-          *     The remaining inflation will be sent to the `eosio.saving` account.
+          *     The remaining inflation will be sent to either `eosio.saving` or `eosio.distrb` (if compiled with USE_INFLATION_DISTRIB).
```
Comment (DenisCarriere) [inline contracts/eosio.system/include/eosio.system/eosio.system.hpp]: Formatting to:

`eosio.dist` & `USE_INFLATION_DISTRIBUTE`

File: contracts/eosio.system/include/eosio.system/eosio.system.hpp (hunk 1 above)
Comment (RalfWeinand) [reply contracts/eosio.system/include/eosio.system/eosio.system.hpp]: I agree `eosio.dist` is probably a better name for it. Apart from that if we keep the existing name then at least we need to fix up the spelling (change from "eosio.ditrib" to "eosio.distrib").

File: contracts/eosio.system/include/eosio.system/eosio.system.hpp (hunk 3 above)
Comment (RalfWeinand) [reply contracts/eosio.system/include/eosio.system/eosio.system.hpp]: For consistency either change to `eosio.dist` (or fix the spelling of `eosio.distrb` to `eosio.distr**i**b`)

**Review by DenisCarriere — APPROVED** [review]

**Pull request closed without merging**